code/game/objects/items/weapons/grenades/ghettobomb.dm

```diff
@@ -30,7 +30,9 @@
 	var/assembled = 0
 	active = 1
 	det_time = 50
-
+	var/list/shrapnel_list = new()
+	var/max_shrapnel = 8
+	var/current_shrapnel = 0
```

Comment:
I haven't read through everything yet, but this var seems unnecessary. Couldn't you just check the length of `shrapnel_list`?